crates/hyperqueue/src/server/job.rs

```diff
@@ -185,16 +187,65 @@ impl Job {
         }
     }
 
-    pub fn make_job_detail(&self, include_tasks: bool) -> JobDetail {
-        let tasks = if include_tasks {
-            self.tasks.values().cloned().collect()
-        } else {
-            Vec::new()
+    pub fn make_job_detail(
+        &self,
+        task_id_selector: IdSelector,
+        task_status_selector: StatusSelector,
```

Comment:
Let's also avoid cloning the selectors here, this function can be called a lot of times and the selector can contain a vector.